rust/src/delta.rs

```diff
@@ -540,14 +556,49 @@ impl DeltaTable {
             self.version(),
         );
 
-        while let PeekCommit::New(new_version, actions) =
-            self.peek_next_commit(self.version()).await?
+        // update to latest version if given max_version is not larger than current version
+        let max_version = max_version.filter(|x| x > &self.version());
+        let max_version: i64 = match max_version {
+            Some(x) => x,
+            None => self.get_latest_version().await?
+        };
+
+        let buf_size = self.config.log_buffer_size;
+
+        // construct stream yielding (version, bytes)
+        let store = self.storage.clone();
+        let log_stream = futures::stream::iter(self.version() + 1..).map(|i| {
+            let store2 = store.clone();
+            let loc = commit_uri_from_version(i).clone();
+            async move {
+                let out = store2.get(&loc);
+                (i, out.await)
+            }
+        });
+        
+        let mut log_buffer = {
+            let n_commits = usize::try_from(max_version - self.version());
+            match n_commits {
+                Ok(n) => log_stream.take(n).buffered(buf_size),
+                Err(err) => return Err(DeltaTableError::GenericError { source: Box::new(err) })
+            }
+        };
```

Comment:
This seems a little complicated. Why couldn't the original iterator be `self.version() + 1..max_version`? And you can just early return if `self.version() >= max_version`?